src/permissions.rs

```diff
@@ -78,7 +80,12 @@ impl TryFrom<&str> for UserIds {
             )));
         };
 
-        Ok(UserIds { uid, gid })
+        Ok(UserIds {
+            user: user.to_string(),
+            group: group.to_string(),
```

Comment:
I don't see `group` field is used anywhere.

Also just an idea for refactoring: could this struct `UserIds` simply hold `libc::group` and `libc::passwd`?

Reply:
OK, I'll change it